source/dub/package_.d

```diff
@@ -796,7 +796,13 @@ private string determineVersionWithGIT(NativePath path)
 	import dub.semver;
 
 	auto git_dir = path ~ ".git";
-	if (!existsFile(git_dir) || !isDir(git_dir.toNativeString)) return null;
+	if (!existsFile(git_dir)) return null;
+	if (!isDir(git_dir.toNativeString)) {
+		auto gitredirect = readText(git_dir.toNativeString).strip();
```

Comment:
If invoking `git` is not viable (e.g. too slow on Windows), then code that messes with git's internals ought to be encapsulated in its own module.

Reply:
Commit message should say that this is for absorbed submodules (i.e. those for which `.git` is a file).

Reply:
Could you please paste your IRC comment for how to do this here? My backlog doesn't go that far.

Reply:
I think this is the relevant line: `[2019-11-22 16:07:24.606] <CyberShadow> git -C path/to/submodule rev-parse --git-dir`

Reply:
Fixed? I've left the Windows version inside the module because I don't think that that refactoring is in-scope for this PR, though I agree it should be done.

Reply:
Fixed via separate PR.